openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -344,29 +345,75 @@ def post(self, request):
         Marks the user as having no password set for deactivation purposes,
         and logs the user out.
         """
-        username = None
         user_model = get_user_model()
         try:
             # Get the username from the request and check that it exists
-            username = request.data['username']
-            user = user_model.objects.get(username=username)
-
+            verify_user_password_response = self._verify_user_password(request)
+            if verify_user_password_response.status_code != status.HTTP_204_NO_CONTENT:
+                return verify_user_password_response
             with transaction.atomic():
                 # 1. Unlink LMS social auth accounts
-                UserSocialAuth.objects.filter(user_id=user.id).delete()
+                UserSocialAuth.objects.filter(user_id=request.user.id).delete()
                 # 2. Change LMS password & email
-                user.email = get_retired_email_by_email(user.email)
-                user.save()
-                _set_unusable_password(user)
+                request.user.email = get_retired_email_by_email(request.user.email)
+                request.user.save()
+                _set_unusable_password(request.user)
                 # 3. Unlink social accounts & change password on each IDA, still to be implemented
+                # 4. Add user to retirement queue
+                UserRetirementStatus.create_retirement(request.user)
```

Comment:
Just double-checking - is it the case that `transaction.atomic()` will rollback *nested* commits (for example, [the row creation in `create_retirement`](https://github.com/edx/edx-platform/blob/1828bf7d6b58470fb42ada3db5ac58bb7bf0d02c/openedx/core/djangoapps/user_api/models.py#L246))?

Reply:
Yes, it will. It will roll back every database transaction executed in this code block if any exception occurs at any point. https://docs.djangoproject.com/en/dev/topics/db/transactions/#django.db.transaction.atomic